GitUI/UserControls/FileStatusList.cs

```diff
@@ -620,91 +621,129 @@ public void SelectStoredNextIndex(int defaultIndex = -1)
+            }
+
+            if (!AppSettings.ShowDiffForAllParents)
+            {
+                tuples.Capacity = 1;
+            }
+
+            // Set summaries and file statuses, if not calculated
+            for (int i = 0; i < tuples.Count; i++)
+            {
+                var par = tuples[i];
+                if (par.summary.IsNullOrWhiteSpace())
                 {
-                    // No parent, will set null as parent
-                    tuples.Add((null, string.Empty, Module.GetTreeFiles(Revision.TreeGuid, full: true)));
+                    par.summary = _diffWithParent.Text + GetDescriptionForRevision(par.parentRev.ObjectId);
                 }
-                else
-                {
-                    if (!AppSettings.ShowDiffForAllParents)
-                    {
-                        parentRevs = new[] { parentRevs[0] };
-                    }
 
-                    foreach (var rev in parentRevs)
-                    {
-                        var desc = (rev.ObjectId != baseRevGuid ? _diffWithParent.Text : _diffWithBase.Text)
-                                   + GetDescriptionForRevision(rev.ObjectId);
-                        var status = Module.GetDiffFilesWithSubmodulesStatus(rev.ObjectId, Revision.ObjectId, Revision.FirstParentGuid);
-                        tuples.Add((rev, desc, status));
-                    }
-
-                    // Show combined (merge conflicts) only when all first (A) are parents to selected (B)
-                    // (a single merge commit is selected with parents explicit or implicit selected)
-                    var isMergeCommit = AppSettings.ShowDiffForAllParents &&
-                                        Revision.ParentIds != null && Revision.ParentIds.Count > 1 &&
-                                        _revisionTester.AllFirstAreParentsToSelected(parentRevs.Select(item => item.ObjectId), Revision);
-                    if (isMergeCommit)
-                    {
-                        var conflicts = Module.GetCombinedDiffFileList(Revision.Guid);
-                        if (conflicts.Count != 0)
-                        {
-                            // Create an artificial commit
-                            var desc = CombinedDiff.Text;
-                            tuples.Add((new GitRevision(ObjectId.CombinedDiffId), desc, conflicts));
-                        }
-                    }
+                if (par.statuses == null)
+                {
+                    par.statuses = Module.GetDiffFilesWithSubmodulesStatus(par.parentRev.ObjectId,
+                        selectedRev.ObjectId,
+                        selectedRev.FirstParentGuid);
                 }
+
+                tuples[i] = par;
             }
 
             GitItemStatusesWithDescription = tuples;
 
             return;
 
-            ObjectId GetRevisionOrHead(ObjectId rev, Lazy<ObjectId> head)
+            ObjectId GetRevisionOrHead(GitRevision rev, Lazy<ObjectId> head)
```

Comment:
```suggestion
            static ObjectId GetRevisionOrHead(GitRevision rev, Lazy<ObjectId> head)
```

Reply:
Re# suggests this (I had that in #7899 but reverted it as it is not used elsewhere in GE). Any motivation you know about? Re# is most often correct though.

Reply:
I guess `static` was not supported before C#8. IIUIC it indicates that no closure is required, i.e. no variables are bound. It just works on the arguments provided.